spark/src/main/scala/org/apache/spark/sql/delta/Checksum.scala

```diff
@@ -213,6 +209,10 @@ trait RecordChecksum extends DeltaLogging {
     // Incrementally compute the new version checksum, if the old one is available.
     val ignoreAddFilesInOperation =
       RecordChecksum.operationNamesWhereAddFilesIgnoredForIncrementalCrc.contains(operationName)
+    val protocol =
+      actions.collectFirst { case p: Protocol => p }.getOrElse(oldVersionChecksum.protocol)
```

Comment:
If the CRC file is missing Protocol, can `oldVersionChecksum.protocol` be null?

Same below for Metadata?